File: backend/api/src/main/java/com/everpath/api/entity/UserProgressEntity.java
package com.everpath.api.entity;

import jakarta.persistence.*;
import lombok.*;

/**
 * Entidad encargada de almacenar
 * el progreso global del usuario.
 *
 * Actualmente solo persiste XP.
 *
 * El nivel se calcula dinámicamente
 * a partir de la experiencia acumulada.
 */
@Entity
@Table(
        name = "user_progress"
)
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class UserProgressEntity {

    @Id
    @GeneratedValue(
            strategy = GenerationType.IDENTITY
    )
    private Long id;


    @OneToOne(
            fetch = FetchType.LAZY,
            optional = false
    )
    @JoinColumn(
            name = "user_id",
            nullable = false,
            unique = true
    )
    private UserEntity user;


    @Column(
            nullable = false
    )
    private Integer xp;
}

File: android-app/app/src/main/java/com/everpath/data/local/entity/UserProgressEntity.kt
package com.everpath.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

/**
 * Entidad Room encargada de
 * persistir el progreso global
 * del usuario.
 */
@Entity(
    tableName = "user_progress"
)
data class UserProgressEntity(

    @PrimaryKey
    val id: Int = 1,

    val xp: Int

)